src/marlin_stubs/host/M46.cpp

```diff
@@ -14,6 +14,8 @@ void GcodeSuite::M46() {
     if (parser.seen('M')) {
         mac_address_t mac;
         get_MAC_address(&mac, netdev_get_active_id());
+        // Append a newline character to the mac parameter
+        strcat(mac, "\n");
```

Comment:
Prove be wrong, but this seems not to fit into the buffer. Looking at mac_addr_t, it's 18 chars long. The MAC address is 17 chars long, you add `\n` and then there's `\0` that overflows now.

Also, `strcat` is the unfortunate combination of unsafe (doesn't check the size) and linear walk through the whole string. Can I ask for either placing the `\n` and `\0` at the right positions manually or using `strlcat`? (yes, we prefer the one with `l` over `strncat`).

Reply:
Sure :) Let me check on this one

Reply:
I'm also not really happy about the strcat and also about the sprintf without size check on the ip and the mac. So I've come up with a different approach using snprintf instead.